core-patch/src/main/scala/eu/ostrzyciel/jelly/core/patch/PatchDecoder.scala

```diff
@@ -0,0 +1,41 @@
+package eu.ostrzyciel.jelly.core.patch
+
+import eu.ostrzyciel.jelly.core.proto.v1.patch.*
+
+import scala.annotation.experimental
+
+/**
+ * Decoder for RDF-Patch streams.
+ *
+ * Converts RdfPatchRow and RdfPatchFrame to callbacks on the given
+ * [[eu.ostrzyciel.jelly.core.patch.handler.PatchHandler]].
+ *
+ * @since 2.11.0
+ */
+@experimental
+trait PatchDecoder:
+  /**
+   * RdfPatchOptions for this decoder.
+   * @return options
+   */
+  def getPatchOpt: Option[RdfPatchOptions]
+
+  /**
+   * Ingest a row into the decoder.
+   *
+   * If the stream has type PUNCTUATED, and the row contains a punctuation mark, the decoder will
+   * call the punctuation() method on the handler.
```

Comment:
don't really get what you mean here. This is a trait that only specifies how Patch Decoders should look like (so any decoders implementing it will have to provide an ingestRow implementation. So the decoder SHOULD call the punctuation() method on the handler, but won't necessarily.

Reply:
No, it will. All implementations must behave the same in this regard. The different impls are only for different statement types.